File: Serene/Serene.Script/Imports/ServerImports/Northwind.CategoryForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    public partial class CategoryForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Category";

        public CategoryForm(string idPrefix) : base(idPrefix) {}

        public StringEditor CategoryName { [InlineCode("{this}.w('CategoryName', Serenity.StringEditor)")] get; private set; }
        public StringEditor Description { [InlineCode("{this}.w('Description', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Imports/ServerTypings/Northwind.CategoryForm.ts
namespace Serene.Northwind {
    export interface CategoryForm {
        CategoryName: Serenity.StringEditor;
        Description: Serenity.StringEditor;
    }

    export class CategoryForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Category';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!CategoryForm.init)  {
                CategoryForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;

                Q.initFormType(CategoryForm, [
                    'CategoryName', w0,
                    'Description', w0
                ]);
            }
        }
    }
}

